addons/web_editor/controllers/main.py

```diff
@@ -607,7 +607,14 @@ def shape(self, module, filename, **kwargs):
                     or attachment.type != 'binary'
                     or not attachment.public
                     or not attachment.url.startswith(request.httprequest.path)):
-                raise werkzeug.exceptions.NotFound()
+                attachment = request.env['ir.attachment'].sudo().search([
+                    '|',
+                        ('url', '=like', request.httprequest.path),
+                        ('url', '=like', '%s?%%' % request.httprequest.path),
+                    ('public', '=', True),
+                ], limit=1)
+                if not attachment:
+                    raise werkzeug.exceptions.NotFound()
```

Comment:
- This needs comments explaining the why in both the commit message and in the code. Something like "This is done to allow to import shape into the system via data files" or something clearer :shrug: 
- Does the usecase with the query string in the URL really is useful? (it was not allowed in 14.0). If yes, maybe it is worth a comment too?
- The `if` above ensures the found attachment is of binary type, should not this be part of the search here too?